besu/src/main/java/org/hyperledger/besu/cli/BesuCommand.java

```diff
@@ -1970,15 +1984,28 @@ private Path pluginsDir() {
     }
   }
 
-  public File nodePrivateKeyFile() {
-    File nodePrivateKeyFile = null;
-    if (isFullInstantiation()) {
-      nodePrivateKeyFile = standaloneCommands.nodePrivateKeyFile;
+  @VisibleForTesting
+  NodeKey buildNodeKey() {
+    return new NodeKey(securityModuleProvider());
+  }
+
+  private SecurityModule securityModuleProvider() {
+    if (DEFAULT_SECURITY_MODULE_PROVIDER.equals(securityModuleProviderName)) {
+      // directly load KeyPairSecurityModule
+      return new KeyPairSecurityModule(KeyPairUtil.loadKeyPair(nodePrivateKeyFile()));
     }
 
-    return nodePrivateKeyFile != null
-        ? nodePrivateKeyFile
-        : KeyPairUtil.getDefaultKeyFile(dataDir());
+    return securityModuleService
+        .getByName(securityModuleProviderName)
+        .orElseThrow(
+            () -> new RuntimeException("Security Module not found: " + securityModuleProviderName))
+        .create(pluginCommonConfiguration);
+  }
+
+  private File nodePrivateKeyFile() {
+    final Optional<File> nodePrivateKeyFile =
+        isDocker ? Optional.empty() : Optional.ofNullable(standaloneCommands.nodePrivateKeyFile);
```

Comment:
There should no be any docker specific behaviour

Reply:
This whole file is using docker specific behavior using org/hyperledger/besu/cli/BesuCommand.java:2103 

~~~
  private boolean isFullInstantiation() {
    return !isDocker;
  }

~~~

Reply:
It's garbage code to be deleted. We don't provide the parameter in the docker image since https://github.com/hyperledger/besu/commit/29c3c142633ce5c4be2dd54812259d99472bd5d1#diff-ebacf6f6ae4ee68078bb16454b23247dL19 so this part of the code is dead since July 2019. It was discussed about removing it but it was not a priority. Now, if it's confusing, it should be a priority IMO.

Reply:
👍 I have created https://github.com/hyperledger/besu/issues/785 and will clean it out in a separate PR.